Packages/Core/Runtime/Variables/AtomVariable.cs

```diff
@@ -51,10 +54,19 @@ public abstract class AtomVariable<T, E1, E2> : AtomBaseVariable<T>
         /// </summary>
         public E2 ChangedWithHistory;
 
+        /// <summary>
+        /// When setting the value of a Variable the new value will be piped through all the pre change transformers, which allows you to create custom logic and restriction on for example what values can be set for this Variable.
+        /// </summary>
+        [SerializeField]
+        public List<F> PreChangeTransformers;
+
         protected abstract bool AreEqual(T first, T second);
 
         private void OnEnable()
         {
+            if (PreChangeTransformers == null) { PreChangeTransformers = new List<F>(); }
```

Comment:
this is always false, because its filled by serialization. also: just initialize a new list at the declaration, to be sure it exists

Reply:
Fixed in the latest commit.